isa/kernel.org/instruction-set.rst

```diff
@@ -11,198 +27,350 @@ all of which are 64-bits wide.
 
 The eBPF calling convention is defined as:
 
- * R0: return value from function calls, and exit value for eBPF programs
- * R1 - R5: arguments for function calls
- * R6 - R9: callee saved registers that function calls will preserve
- * R10: read-only frame pointer to access stack
+* R0: return value from function calls, and exit value for eBPF programs
+* R1 - R5: arguments for function calls
+* R6 - R9: callee saved registers that function calls will preserve
+* R10: read-only frame pointer to access stack
```

Comment:
Is the VM supposed to enforce this? Note: neither bpf2c nor uBPF enforces this restriction.

Reply:
@Alan-Jowett can you clarify what you mean by "this"?

Reply:
Got it, updated

Reply:
Actually I think the verifier should disallow modification of R10.   @qmonnet can you check what Linux does here?

Reply:
Yes, R10 has always been read-only. Since the verifier was introduced in Linux, it has been ensuring that R10 is never passed as the destination register (returning `-EACCES` otherwise). Some other checks rely on this assumption.

Reply:
Thanks for verifying, I will make sure the PREVAIL verifier does the same.